### PR TITLE
net: Fix skb allocation

### PR DESCRIPTION
**Pull request by alexkalmuk**
These changes revert 3107afe030b1c56e82cf648d2d3a977238401610 which is partially wrong since all huge packets are dropped.

This also fixes template x86/qt-vnc

### CODE REVIEW

File: src/net/skbuff/skb_data.c [hunk 1]
```diff
@@ -125,9 +139,16 @@ void skb_data_free(struct sk_buff_data *skb_data) {
 	sp = ipl_save();
 	{
 		if (--skb_data->links == 0) {
-			assert(pool_belong(&skb_data_pool, skb_data));
-
-			pool_free(&skb_data_pool, skb_data);
+			assert(skb_data->alloc_type == ALLOCATED_POOL ||
+					skb_data->alloc_type == ALLOCATED_MALLOC);
+			switch (skb_data->alloc_type) {
+			case ALLOCATED_POOL:
```
Comment (anton-bondarev) [inline src/net/skbuff/skb_data.c]: default: assert()

Comment (alexkalmuk) [reply src/net/skbuff/skb_data.c]: Thanks, fixed.

File: src/net/skbuff/skb_data.c [hunk 2]
```diff
@@ -125,9 +139,17 @@ void skb_data_free(struct sk_buff_data *skb_data) {
 	sp = ipl_save();
 	{
 		if (--skb_data->links == 0) {
-			assert(pool_belong(&skb_data_pool, skb_data));
-
-			pool_free(&skb_data_pool, skb_data);
+			switch (skb_data->alloc_type) {
+			case ALLOCATED_POOL:
```
Comment (AntonKozlov) [inline src/net/skbuff/skb_data.c]: Another option is to use `pool_belong`, in this case no need to store allocation metainfo

Comment (alexkalmuk) [reply src/net/skbuff/skb_data.c]: Yes, that's exactly like it was previously done, but as for me, it is more safely and clearly to state explicitly how the object is allocated.

**Pull request merged**